packages/client-adapters/src/adapters/lexicons/adapter.js

```diff
@@ -130,6 +131,68 @@ class AlpheiosLexiconsAdapter extends BaseAdapter {
       this.addError(this.l10n.messages.LEXICONS_NO_ALLOWED_URL)
       return
     }
+
+    if (this.async) {
+      return this.fetchDefsWithCallbacks(homonym, lookupFunction)
+    } else {
+      if (lookupFunction === 'short') {
+        return this.fetchShortDefsSync(homonym)
+      } else if (lookupFunction === 'full') {
+        await this.fetchFullDefsSync(homonym)
+      }

```

Comment:
Why to we return short defs without waiting for the result to arrive while awaiting on full definitions? Shouldn't it be handle in the same manner?

Reply:
good catch - i have missed this - fixed